ensuring that tags does not show in branches

callmeradical opened the pull request:
Fixes issue #63 tags shows as a branch when issuing 
```bash
git name-rev --name-only HEAD
```
On a tagged commit

Commit:
ensuring that tags does not show in branches

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/harbur/captain?pullRequest=65) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/harbur/captain?pullRequest=65) before we can accept your contribution.<br/>

dkapanidis committed:
Added user-defined tag

This is quite useful for Continuous Delivery use-cases.

In order for CI systems to propagate the BUILD_NUMBER to captain they can now use '-t BUILD_NUMBER' as parameter.
This will create a docker image with the specific version and push it to registry with 'captain push'

spiddy commented on git.go:
```diff
@@ -33,7 +33,9 @@ func getBranches(all_branches bool) []string {
 
 		// Branches list is separated by spaces. Let's put it in an array
-		labels = append(labels, branches...)
+		if !strings.Contains(branches_str, "tags") {
```
If I create a branch named "catagsarm" it will be omitted this way because it contains 'tags' inside the text.

What is your current use-case? Does your git spit your tags as branches?

callmeradical replied:
Hey it's updated now, see how that looks?

dkapanidis reviewed (changes requested):
It should avoid omitting branches that may include 'tags' inside their name

dkapanidis committed:
Preparing version 1.1.0

Commit:
checking test code

dlozano committed:
Exit after test failure

Commit:
adding mock in for test for tags

Commit:
Merge branch 'master' into master

Commit:
Preparing version 1.1.1p

jwliechty commented:
This is still an issue when using git 2.17.1. (from a fresh install of Ubuntu 18.04)
Is there a way to move this forward?
